src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -0,0 +1,273 @@
+import React, {useEffect, useState, useRef, useCallback} from 'react';
+import {View, InteractionManager} from 'react-native';
+import PropTypes from 'prop-types';
+import lodashGet from 'lodash/get';
+import _ from 'underscore';
+import {useFocusEffect} from '@react-navigation/native';
+import styles from '../../../styles/styles';
+import BigNumberPad from '../../../components/BigNumberPad';
+import * as CurrencyUtils from '../../../libs/CurrencyUtils';
+import * as MoneyRequestUtils from '../../../libs/MoneyRequestUtils';
+import Button from '../../../components/Button';
+import * as DeviceCapabilities from '../../../libs/DeviceCapabilities';
+import TextInputWithCurrencySymbol from '../../../components/TextInputWithCurrencySymbol';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+import HeaderWithBackButton from '../../../components/HeaderWithBackButton';
+import useLocalize from '../../../hooks/useLocalize';
+import CONST from '../../../CONST';
+
+const propTypes = {
+    /** IOU amount saved in Onyx */
+    amount: PropTypes.number,
+
+    /** Currency chosen by user or saved in Onyx */
+    currency: PropTypes.string,
+
+    /** Title to be displayed in the header */
+    title: PropTypes.string,
+
+    /** Whether the amount is being edited or not */
+    isEditing: PropTypes.bool,
+
+    /** Fired when back button pressed, navigates back to a proper page */
+    onBackButtonPress: PropTypes.func.isRequired,
+
+    /** Fired when back button pressed, navigates to currency selection page */
+    onCurrencyButtonPress: PropTypes.func.isRequired,
+
+    /** Fired when submit button pressed, saves the given amount and navigates to the next page */
+    onSubmitButtonPress: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    amount: 0,
+    currency: CONST.CURRENCY.USD,
+    title: '',
+    isEditing: false,
+};
+
+const amountViewID = 'amountView';
+const numPadContainerViewID = 'numPadContainerView';
+const numPadViewID = 'numPadView';
+
+function MoneyRequestAmountForm(props) {
```

Comment:
We should use prop destructuring -> https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring